app/controllers/floors_controller.rb

```diff
@@ -1,29 +1,28 @@
 class FloorsController < ApplicationController
   before_action :set_floors, only: %i[index]
   before_action :set_floor, only: %i[update]
+  before_action :show_navbar
   # before_action :set_locations, only: %i[update]
 
   def index
-    @floors = @floors.order('level ASC')
-    @floor = params[:floor_number] || 2
-    @toggle_nav = params[:toggle_navbar] || false
-    @maps = extract_maps_from_floors(@floors)
-    @persistent_locations = fetch_persistent_locations
-    @icons_images = extract_icon_images
-    @search_results = process_search(params[:search]) if params[:search]
-    if current_user
-      @admin_user = current_user.admin?
-    else
-      @admin_user = false
-    end
-    @search_results = @search_results.flatten if @search_results
-    if @search_results
-      @locations = extract_locations(@search_results) if @search_results
-      @search_result_floors = extract_floors(@locations) if @locations
-      @edit_locations = Location.all
-    else
-      @edit_locations = Location.all
-    end
+    search_results = process_search(params[:search])
+    locations = extract_locations(search_results)
+    search_result_floors = extract_floors(locations)
+    admin_user = false
+    admin_user = current_user.admin? if user_signed_in?
+
+    @state = {
```

Comment:
Discussed previously. This is a major refactor to reflect the react portion of the app in rails. When new data is added, this makes it easier to add the data. Instead of editing 3 files, we edit one and the new data is in the react app.